generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/PropertyDataFetcher.kt

```diff
@@ -16,22 +16,36 @@
 
 package com.expediagroup.graphql.generator.execution
 
-import graphql.TrivialDataFetcher
 import graphql.schema.DataFetcher
 import graphql.schema.DataFetchingEnvironment
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.LightDataFetcher
+import java.util.function.Supplier
 import kotlin.reflect.KProperty
 
 /**
  * Property [DataFetcher] that directly invokes underlying property getter.
  *
- * @param propertyGetter Kotlin property getter that will be invoked to resolve a field
+ * @param propertyGetter Kotlin's property getter that will be invoked to resolve a field
  */
-class PropertyDataFetcher(private val propertyGetter: KProperty.Getter<*>) : TrivialDataFetcher<Any?> {
+class PropertyDataFetcher(private val propertyGetter: KProperty.Getter<*>) : LightDataFetcher<Any?> {
```

Comment:
`PropertyDataFetcher` will now implement the  [LightDataFetcher](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/schema/LightDataFetcher.java#L14), which will make the execution engine to create the **dataFetchingEnviroment** on demand, most (if not all), PropertyDataFetchers dont require this field, so this will represent a memory improvement